### PR TITLE
WebGLRenderer: Avoid default color space conversion.

### LINKED ISSUE #21318: TextureEncodingTest.gltf is different from expected result
**Describe the bug**

I tried Khronos' new test model [TextureEncodingTest.gltf](https://github.com/KhronosGroup/glTF-Sample-Models/tree/encoding/2.0/TextureEncodingTest) with Three.js.
However, it seems to be different from the expected result.

**To Reproduce**

[Three.js r125 + TextureEncodingTest.gltf result:](https://cx20.github.io/gltf-test/examples/threejs/index.html?category=tutorialModels&model=TextureEncodingTest&scale=0.2&type=glTF)
image

**Expected behavior**

image

**Platform:**

- Device: MacBook Air
- OS: Windows 10 (BootCamp)
- Browser: Chrome 88.0.4324.182
- Three.js version: r125

### PR DESCRIPTION
**Pull request by takahirox**
Related issue: #21318, #21323, #21324

**Description**

This PR is an alternative of #21323 and #21324, and resolves #21318. Please refer to them for the detail of this change.

### CODE REVIEW

**Comment by mrdoob** [comment]
Thanks!

**Comment by Mugen87** [comment]
@mrdoob I wonder if `UNPACK_COLORSPACE_CONVERSION_WEBGL` was the root cause for #20786. Maybe RGBM files can now be loaded correctly on iOS even with `TextureLoader` 🤔 ?

TBH, `UNPACK_COLORSPACE_CONVERSION_WEBGL` confused me quite a bit. I'm a bit surprised that color textures were decoded from srgb to linear not twice in the past. 

The standard also mentioned that it is up to the browser if a conversion is applied on how the conversion is implemented. Sounds quite unreliable. On reason more to disable it.

**Comment by lexaknyazev** [comment]
> were decoded from srgb to linear not twice in the past

AFAIR, all that setting does is comparing the image file profile (when present) with some "system default" profile (e.g. sRGB), performing color-matching, and (re)encoding corrected values (usually to 8-bit sRGB).

**Comment by Mugen87** [comment]
Related issue: https://stackoverflow.com/questions/22679494/three-js-normal-map-rendering-differently-windows-mac

**Comment by Mugen87** [comment]
> AFAIR, all that setting does is comparing the image file profile (when present) with some "system default" profile (e.g. sRGB), performing color-matching, and (re)encoding corrected values (usually to 8-bit sRGB).

Okay, that would at least explain why the sRGB decoding works as expected in the past. But the stackoverflow thread and the RGBM issue on iOS seems to be related to `UNPACK_COLORSPACE_CONVERSION_WEBGL`.

**Comment by Mugen87** [comment]
This is the RGBM example with the dev build and `TextureLoader`. Can somebody with an iOS device check if the same banding occurs like in #20786?

https://jsfiddle.net/6hjtzy8x/

**Comment by mrdoob** [comment]
@Mugen87 I've made a copy of the jsfiddle to glitch. On Chrome I can see the white getting darker:

https://mrdoob-sandbox.glitch.me/webgl_loader_texture_rgbm.html
https://raw.githack.com/mrdoob/three.js/dev/examples/webgl_loader_texture_rgbm.html

On Safari... Well, I just realised all the examples are broken in Safari right now. Investigating...

**Comment by mrdoob** [comment]
Okay, found the reason Safari is broken.
And after fixing that localy, I get this on Safari/macOS already:

Screen Shot 2021-02-22 at 8 51 59 PM

**Comment by lexaknyazev** [comment]
@mrdoob 
> https://mrdoob-sandbox.glitch.me/webgl_loader_texture_rgbm.html
> https://raw.githack.com/mrdoob/three.js/dev/examples/webgl_loader_texture_rgbm.html

Is it intentional that different PNG images are used in these two samples?

**Comment by mrdoob** [comment]
Oh sorry. Let me try again.

**Comment by mrdoob** [comment]
I don't think that matters for what we're investigating here.

**Comment by Mugen87** [comment]
Okay, then #20786 is indeed unrelated to `UNPACK_COLORSPACE_CONVERSION_WEBGL`. After some more testing, it's about premultiplying alpha which behaves differently on iOS. So we definitely need a custom RGBM loader.

**Comment by mrdoob** [comment]
@Mugen87 It's worth noting that Safari Technical Preview seem to have this issue fixed:

Screen Shot 2021-02-23 at 2 07 46 PM

So I guess we'd just need the workaround until summer?

**Comment by Mugen87** [comment]
Yes, probably. Maybe revisiting this issue in second half of 2021?

**Comment by takahirox** [comment]
Does anyone here know if embedded custom color space is often used in practical? I'm curious to know how much this change affects the existing Three.js applications. And this is a breaking change so I'm going to write a note that we recommend pre apply the transformation in Migration guide.

**Comment by lexaknyazev** [comment]
Based on my experience, here are the most often cases:
- a regular sRGB profile embedded just in case (safe to ignore on the web);
- a wide gamut profile is embedded in screenshots taken on P3 displays on macOS (ignoring it leads to compressing image colors);
- photos not prepared for web (may contain anything);
- images authored for printing (may even use CMYK for pixel values).

**Comment by takahirox** [comment]
Thanks for the explanation. It sounds like to me that embedded extra image information is beyond the use of web 3D and avoid default color space conversion and request pre apply the color transformation may be a good direction as Web based 3D engine for portability and performance.

I added the note to [Migration Guide](https://github.com/mrdoob/three.js/wiki/Migration-Guide#r125--r126). Please feel free to reword it because English is not my first language.
